File: tests/ma_librairie/ma_librairie/f_src/calcul.f
      integer*4 function for_area(rectptr, area)
         include 'pikendus_types.finc'

         integer*8 :: rectptr
         integer*4, intent(out) :: area

         type(rectangle) :: rect
         pointer(rectptr, rect)
         integer*4 :: tmp

         write(6,*) "[DEBUG]Ceci est un message de debug"

         tmp = (rect%ur_corner%x - rect%ll_corner%x) *
     &         (rect%ur_corner%y - rect%ll_corner%y)

         area = abs(tmp)

         for_area = 0

      end function

File: tests/ma_librairie/ma_librairie/c_src/calcul.c
#include <stdint.h>
#include <math.h>
#include "_pikendus_types.h"


int c_create_rectangle(rectangle *r, uint32_t width, uint32_t height)
{
    r->ll_corner.x=0;
    r->ll_corner.y=0;

    r->ur_corner.x=width;
    r->ur_corner.y=height;

    return 0;

}
